wrapper/src/main/java/software/amazon/jdbc/plugin/limitless/LimitlessConnectionPlugin.java

```diff
@@ -140,27 +157,29 @@ private Connection connectInternal(
       LOGGER.finest(Messages.get("LimitlessConnectionPlugin.limitlessRouterCacheEmpty"));
       final boolean waitForRouterInfo = WAIT_F0R_ROUTER_INFO.getBoolean(props);
       if (waitForRouterInfo) {
-        limitlessRouters = synchronouslyGetLimitlessRoutersWithRetry(props);
+        limitlessRouters = synchronouslyGetLimitlessRoutersWithRetry(connectFunc.call(), hostSpec.getPort(), props);
       } else {
         LOGGER.finest(Messages.get("LimitlessConnectionPlugin.usingProvidedConnectUrl"));
         return connectFunc.call();
       }
-    } else if (limitlessRouters.contains(hostSpec)) {
+    }
+
+    if (limitlessRouters.contains(hostSpec)) {
       LOGGER.finest(Messages.get("LimitlessConnectionPlugin.connectWithHost", new Object[] {hostSpec.getHost()}));
       return connectFunc.call();
     }
 
     RoundRobinHostSelector.setRoundRobinHostWeightPairsProperty(props, limitlessRouters);
-    final HostSpec selectedHostSpec;
+    HostSpec selectedHostSpec;
     try {
       selectedHostSpec = this.pluginService.getHostSpecByStrategy(limitlessRouters,
           HostRole.WRITER, RoundRobinHostSelector.STRATEGY_ROUND_ROBIN);
       LOGGER.fine(Messages.get(
           "LimitlessConnectionPlugin.selectedHost",
           new Object[] {selectedHostSpec.getHost()}));
     } catch (UnsupportedOperationException e) {
-      LOGGER.severe(Messages.get("LimitlessConnectionPlugin.incorrectConfiguration"));
-      throw e;
+      selectedHostSpec = hostSpec;
+      LOGGER.warning(Messages.get("LimitlessConnectionPlugin.errorSelectingRouter", new Object[] {e.getMessage()}));
```

Comment:
Such exception means that requested strategy isn't loaded. I think it's a showstopper.

Reply:
You also need to catch SQLException. Some strategies raises SQLException when they can't get a host.